xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,77 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
 }
 
 DEFINE_TESTCASE(replace_weights, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
 
     Xapian::Database mydb(get_database("apitest_onedoc"));
 
     Xapian::Enquire enquire(mydb);
 
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top ten results (we only expect one)
+    // retrieve the top ten results (we only expect one).
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
-    // assigning new weight of -1 to the document.
+    /* assigning new weight of -1 to the document.
+     * a negative weight was chose to avoid matching the positive weight returned by enquire.
```

Comment:
Yes, let's avoid negative weights.  Negative weights should perhaps be rejected if you try to set them, though that raises issues with the `MSet` weights being partially updated when we discover a negative one.

If you want a weight that's definitely lower than the maximum attained weight, I'd suggest just halving an existing weight.